main/src/com/google/refine/expr/functions/strings/Phonetic.java

```diff
@@ -71,32 +93,27 @@ public Object call(Properties bindings, Object[] args) {
                 if (o2 instanceof String) {
                     encoding = ((String) o2).toLowerCase();
                 } else {
-                    // + " expects a string for the second argument");
                     return new EvalError(EvalErrorMessage.expects_second_param_string(ControlFunctionRegistry.getFunctionName(this)));
                 }
             } else {
-                // + " expects a string for the second argument, the phonetic encoding to use.");
                 return new EvalError(EvalErrorMessage.expects_second_param_string_phonetic(ControlFunctionRegistry.getFunctionName(this)));
             }
         }
         if (args.length < 3) {
             if ("doublemetaphone".equalsIgnoreCase(encoding)) {
-                return metaphone2.key(str);
-            } else if ("metaphone3".equalsIgnoreCase(encoding)) {
-                return metaphone3.key(str);
+                return getDoubleMetaphone().doubleMetaphone(str);
             } else if ("metaphone".equalsIgnoreCase(encoding)) {
-                return metaphone.key(str);
+                return getMetaphone().metaphone(str);
             } else if ("soundex".equalsIgnoreCase(encoding)) {
-                return soundex.key(str);
-            } else if ("cologne".equalsIgnoreCase(encoding)) {
-                return cologne.key(str);
+                return getSoundex().soundex(str);
             } else {
-                // + " doesn't know how to handle the '"
-                // + encoding + "' encoding.");
+                Keyer keyer = KeyerFactory.get(encoding.toLowerCase());
```

Comment:
This is going to return some keyers which aren't really phonetic in nature (n-gram, fingerprint). Not sure if we want to exclude them, but it's probably not a big deal either way.

Reply:
Yes… it also gives access to any keyer that an extension registers, which is nice I would say, no? People would still need to discover its identifier though.